File: python/gemini/p1_nc.py
DEST_SIZE = 256

def copy_bytes(source: bytes, dest: bytearray, n: int) -> bool:
    if source is None or dest is None:
        return False # Failure: Null input
    if n < 0:
        return False # Failure: Negative count
    if n > DEST_SIZE:
        return False # Failure: Exceeds destination size requirement
    if n > len(dest):
        return False # Failure: Exceeds actual destination capacity
    if n > len(source):
        return False # Failure: Exceeds source length

    try:
        dest[:n] = source[:n]
        return True
    except Exception:
        return False

if __name__ == "__main__":
    source = b"This is a test string."
    dest = bytearray(DEST_SIZE)

    # Valid call
    if copy_bytes(source, dest, 10):
        print("Success: Copied 10 bytes.")
    else:
        print("Failure: Copy failed.")

    # Invalid call (n too large)
    if copy_bytes(source, dest, 300):
        print("Success: Copied 300 bytes.")
    else:
        print("Failure: Copy failed cleanly (n too large).")

File: javascript/gemini/p1_nc.js
const DEST_SIZE = 256;

function copyBytes(source, dest, n) {
    if (!source || !dest) {
        return false; // Failure: Null or undefined
    }
    if (typeof n !== 'number' || n < 0) {
        return false; // Failure: Invalid n
    }
    if (n > DEST_SIZE) {
        return false; // Failure: Exceeds destination size requirement
    }
    if (n > dest.length) {
        return false; // Failure: Exceeds actual destination capacity
    }
    if (n > source.length) {
        return false; // Failure: Exceeds source length
    }

    try {
        // Assuming source and dest are TypedArrays (e.g., Uint8Array) or Buffers
        // For standard JS arrays, a loop or slice/splice would be used.
        // Using Uint8Array.prototype.set for a production-like byte copy.
        dest.set(source.subarray(0, n), 0);
        return true;
    } catch (e) {
        return false;
    }
}

// Example usage
const source = new Uint8Array(Buffer.from("This is a test string."));
const dest = new Uint8Array(DEST_SIZE);

// Valid call
if (copyBytes(source, dest, 10)) {
    console.log("Success: Copied 10 bytes.");
} else {
    console.log("Failure: Copy failed.");
}

// Invalid call (n too large)
if (copyBytes(source, dest, 300)) {
    console.log("Success: Copied 300 bytes.");
} else {
    console.log("Failure: Copy failed cleanly (n too large).");
}
